sympy/solvers/ode/ode.py

```diff
@@ -1075,11 +1073,19 @@ class in it.  Note that a hint may do this anyway if
     # Any ODE that can be solved with a combination of algebra and
     # integrals e.g.:
     # d^3/dx^3(x y) = F(x)
-    ode = SingleODEProblem(eq_orig, func, x)
-    solver = NthAlgebraic(ode)
-    if solver.matches():
-        matching_hints['nth_algebraic'] = solver
-        matching_hints['nth_algebraic_Integral'] = solver
+    ode = SingleODEProblem(eq, func, x, eq_orig)
+    solvers = {
+        NthAlgebraic: ('nth_algebraic','nth_algebraic_Integral'),
+        FirstLinear: ('1st_linear','1st_linear_Integral'),
+        AlmostLinear: ('almost_linear','almost_linear_Integral'),
+        Bernoulli: ('Bernoulli','Bernoulli_Integral')
```

Comment:
I think it would be better to check the `has_integral` attribute below rather than list them with the `_Integral` hint here.